src/modules/url-preview/generatePreview.js

```diff
@@ -0,0 +1,27 @@
+import winston from 'winston';
+import fetchFromEndPoint from './fetchFromEndPoint';
+import fetchFromMetaData from './fetchFromMetaData';
+
+const strategies = [
+	fetchFromEndPoint,
+	fetchFromMetaData
+];
+
+export default async function (url) {
+	return new Promise(async (resolve) => {
```

Comment:
You don't need to create a new promise here. It's already inside an async function.